### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by pinzart90**
add automated tests

### CODE REVIEW

File: src/Config/CS.props [hunk 1]
```diff
@@ -3,7 +3,6 @@
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <PlatformTarget >x64</PlatformTarget>
-    <TargetFramework>net8.0</TargetFramework>
```
Comment (pinzart90) [inline src/Config/CS.props:6]: added targetframework to each project to be easier to understand

File: src/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<Project>
+  <PropertyGroup>
+    <!--
+      SolutionDir is mostly useful for running the dotnet test command
+      Ex. open a command line in the DynamoSamples directory and run:
+          dotnet test %cd%\src\SampleLibraryTests -p:Configuration=Release - -filter "TestCategory!=Failure" - -logger "trx;LogFileName=results.trx" - -results-directory TestResults
+      Note: The - -filter, - -logger and - -results-directory should have double dashes without a space in between. -->
+    <SolutionDir>$(MSBuildThisFileDirectory)\</SolutionDir>
```
Comment (mjkkirschner) [inline src/Directory.Build.props:8]: nice

File: src/SampleLibraryTests/HelloDynamoZeroTouchTests.cs [hunk 3]
```diff
@@ -44,13 +44,15 @@ namespace SampleLibraryTests
     class HelloDynamoZeroTouchTests : GeometricTestBase
     {
         [Test]
+        [Category("NEEDS_ASM")]
```
Comment (mjkkirschner) [inline src/SampleLibraryTests/HelloDynamoZeroTouchTests.cs]: I am not sure we should call this ASM - maybe geo lib or something less ADSK speak.

**Review by mjkkirschner — APPROVED** [review]
it looks good @pinzart90 - I did run into the issue I mentioned the other day again, some tests failing discovery because Dynamo's binaries are not around at discovery time.

This is especially an issue when writing tests for packages using AssemblyLoadContext isolation because there are likely conflicting dependencies with Dynamo and dumping your test binaries into the DynamoCore folder is going to put you right back into that mess.


I was curious if we have that filed somewhere? (not a blocker for merging this of course)

**Comment by pinzart90** [comment]
> it looks good @pinzart90 - I did run into the issue I mentioned the other day again, some tests failing discovery because Dynamo's binaries are not around at discovery time.
> 
> This is especially an issue when writing tests for packages using AssemblyLoadContext isolation because there are likely conflicting dependencies with Dynamo and dumping your test binaries into the DynamoCore folder is going to put you right back into that mess.
> 
> I was curious if we have that filed somewhere? (not a blocker for merging this of course)

YEs, tracked internally DYN-6591

**Pull request merged**